tms/src/main/java/com/netcracker/edu/tms/dao/ProjectDaoJPAImpl.java

```diff
@@ -0,0 +1,127 @@
+import javax.persistence.*;
+
+@Repository
+public class ProjectDaoJPAImpl implements ProjectDao {
+    private static final Logger LOGGER = LogManager.getLogger();
+
+    @PersistenceContext
+    private EntityManager entityManager;
+
+    @Override
+    public Project getProjectById(BigInteger projectId) {
+        try{
+            LOGGER.info("getEmployeeById called with \""+projectId+"\"");
+            return entityManager.find(Project.class, projectId);
+        }catch (IllegalArgumentException e){
+            LOGGER.warn("IllegalArgumentException in getEmployeeById appeared",e);
+            throw e;
+        }
+    }
+
+    @Override
+    public List<Project> getProjectByName(String name) {
+        try {
+            Query query = entityManager.createQuery(QueryConsts.SELECT_WITH_NAME);
+            query.setParameter("surname", name);
+            List<Project> ret = query.getResultList();
+
+            LOGGER.info("getEmployeesBySurname called with \""+name+"\"");
+            return ret;
+        }catch (IllegalArgumentException e){
+            LOGGER.warn("IllegalArgumentException in getEmployeesBySurname appeared",e);
+            throw e;
+        }
+    }
+
+    @Override
+    public boolean addProject(Project project) {
+        try{
+            entityManager.persist(project);
+            LOGGER.info("addEmployee called with \""+project+"\"");
+            return true;
+        } catch (Exception e) {
+            LOGGER.warn("Some exception appeared in addEmployee",e);
+            throw e;
+        }
+
+    }
+
+    @Override
+    public boolean updateProject(Project project, Project oldProject) {
+        try {
+            if(entityManager.contains(oldProject)){
+                entityManager.merge(project);
+                LOGGER.info("updateProject called with \""+project+"\"");
+
+                return true;
+            }else{
+                LOGGER.warn("No such Project in updateProject appeared");
+                throw new IllegalStateException("No such Project");//Not found entity to update
+            }
+
+        } catch (Exception e) {
+            LOGGER.warn("IllegalArgumentException in updateProject appeared",e);
+            throw e;
+        }
+
+    }
+
+    @Override
+    public boolean deleteProject(Project project) {
+        try {
+            BigInteger identifier=project.getId();
+
+            Project same=entityManager.find(Project.class,identifier);
+            if(same.equals(project)) {
+                entityManager.remove(same);
+                LOGGER.info("deleteProject called with \""+project+"\"");
+                return true;
+            }else{
+                LOGGER.warn("No such Employee in deleteEmpoyee appeared");
+                return false;
+            }
+        } catch (IllegalArgumentException e) {
+            LOGGER.warn("IllegalArgumentException in deleteEmployee appeared",e);
+            throw e;
+        }
+        catch (NullPointerException e) {
```

Comment:
It is not good practise to catch NPE
better to prevent it by NOT NULL checking like if (same != null) {...}